Bottom: 692276f258db83d12b27e6ee0c7536d6c82c3325
Top:    9738db240d331c4f1ad1828699c130e20eb5cfcb
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:27 +0000

Debug stuff


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 1c0add1..d2c80e7 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1234,24 +1234,21 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 
       fprintf (stderr, "Computing copper geometry for group %i\n", group);
 
-#if 1
       GROUP_LOOP (PCB->Data, group);
         {
           fprintf (stderr, "Accumulating elements from layer %i\n", GetLayerNumber (PCB->Data, layer));
 
-//          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
-//          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
-//          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
+          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
+          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
+          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
           r_search (layer->polygon_tree, &bounds, NULL, polygon_copper_callback, &info);
         }
       END_LOOP;
 
-//      fprintf (stderr, "Accumulating pin + via pads\n");
-//      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-//      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
-#endif
+      fprintf (stderr, "Accumulating pin + via pads\n");
+      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 
-#if 0
       if (group == top_group ||
           group == bottom_group)
         {
@@ -1259,7 +1256,6 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
           fprintf (stderr, "Accumulating SMT pads for side %i\n", info.side);
           r_search (PCB->Data->pad_tree, &bounds, NULL, pad_copper_callback, &info);
         }
-#endif
 
       /* TODO: Inter-layer features
        *
diff --git a/src/polygon1.c b/src/polygon1.c
index 039b9b5..b662cb8 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,7 +39,7 @@
       are marked
 */
 
-#undef NDEBUG
+//#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -110,13 +110,13 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#define DEBUG_INTERSECT
-#define DEBUG_LABEL
-#define DEBUG_ALL_LABELS
-#define DEBUG_JUMP
-#define DEBUG_GATHER
-#define DEBUG_ANGLE
-#define DEBUG
+#undef DEBUG_INTERSECT
+#undef DEBUG_LABEL
+#undef DEBUG_ALL_LABELS
+#undef DEBUG_JUMP
+#undef DEBUG_GATHER
+#undef DEBUG_ANGLE
+#undef DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
@@ -2964,9 +2964,13 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
 
   *res = NULL;
 
+#if 0
   /* Make copies for tracking polygon parentage (DEBUG) */
   if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
       return err_no_memory;
+#endif
+  a_copy = NULL;
+  b_copy = NULL;
 
   /* Move the parentage information over onto the copy */
   if (a_copy != NULL)
